odoo/addons/base/models/res_partner.py

```diff
@@ -346,6 +347,13 @@ def _compute_display_name(self):
         for partner in self:
             partner.display_name = names.get(partner.id)
 
+    @api.depends_context('lang')
+    @api.depends('display_name')
+    def _compute_translated_display_name(self):
+        names = dict(self.name_get())
```

Comment:
This is more like `_compute_context_dependent_display_name`. To be conservative maybe `names = dict(self.with_context({'lang': self.env.lang}).name_get()))`